### PR TITLE
Addd subgraph deployment script

### LINKED ISSUE #342: Mosaic subgraph should support templating
Currently, Mosaic subgraphs has contract addresses for goerli and 1405. In order to use same subgraph, subgraph deployment must accept contract addresses as an argument.  

Aim of the ticket is to create subgraph templates and script/command to deploy subgraph given the contract addresses as input.

### PR DESCRIPTION
**Pull request by 0xsarvesh**
fixes #342  
1. Added command to deploy subgraph,
2. subgraph project now supports templating.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,5 @@ openst-facilitator-*.tgz
 # Subgraph build files
 subgraph/auxiliary/build/**
 subgraph/origin/build/**
+`subgraph/origin/subgraph.yaml
```
Comment (abhayks1) [inline .gitignore]:
```suggestion
subgraph/origin/subgraph.yaml
```

File: README.md [hunk 2]
```diff
@@ -285,5 +285,30 @@ Facilitator can be started in below two ways :-
   
   Refer detailed document [here](demo/README.md). 
  
+### Deploy mosaic subgraph
```
Comment (abhayks1) [inline README.md:288]: This readme part we should move to `subgraph` folder.  subgraph folder will move to `mosaic-subgraph` repository.

File: package.json [hunk 3]
```diff
@@ -58,6 +60,7 @@
     "@openst/mosaic-chains": "0.1.0-alpha.9",
     "@openst/mosaic-contracts": "0.12.0",
     "@openst/mosaic-proof": "0.1.0-alpha.1",
+    "@types/mustache": "^4.0.1",
```
Comment (abhayks1) [inline package.json]:
```suggestion
    "@types/mustache": "4.0.1", 😞 
```

File: package.json [hunk 4]
```diff
@@ -84,7 +88,8 @@
     "web3-eth-accounts": "1.0.0-beta.55",
     "web3-utils": "1.0.0-beta.55",
     "winston": "3.2.1",
-    "ws": "7.0.0"
+    "ws": "7.0.0",
+    "@graphprotocol/graph-cli": "0.12.0"
```
Comment (abhayks1) [inline package.json]: Why this package needed in root package.json?

File: subgraph/src/DeployAuxiliarySubgraph.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    params.graphAdminRPCEndpoint,
+    params.ipfsEndpoint,
+    auxiliarySubgraphDirectory,
+    auxiliarySubgraphName,
+  ).deploy({
+    auxiliaryAnchor: params.auxiliaryAnchor,
+    erc20Cogateway: params.erc20Cogateway,
+  });
+} catch (e) {
+  console.log('Error in deployment of subgraph ', e.message);
```
Comment (abhayks1) [inline subgraph/src/DeployAuxiliarySubgraph.ts]: Remove console.log.
Also command should process.exit(1) on error.

File: subgraph/src/DeployOriginSubgraph.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    params.graphAdminRPCEndpoint,
+    params.ipfsEndpoint,
+    originSubgraphDirectory,
+    originSubgraphName,
+  ).deploy({
+    originAnchor: params.originAnchor,
+    erc20Gateway: params.erc20Gateway,
+  });
+} catch (e) {
+  console.log('Error in deployment of subgraph ', e.message);
```
Comment (abhayks1) [inline subgraph/src/DeployOriginSubgraph.ts]: Same comment as above.

File: subgraph/src/DeployAuxiliarySubgraph.ts [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  auxiliaryAnchor: string;
+  erc20Cogateway: string;
+  graphAdminRPCEndpoint: string;
+  ipfsEndpoint: string;
+} {
+  assert(process.argv.length === 6);
+
+  const auxiliaryAnchor = process.argv[2];
+  const erc20Cogateway = process.argv[3];
+  const graphAdminRPCEndpoint = process.argv[4];
```
Comment (abhayks1) [inline subgraph/src/DeployAuxiliarySubgraph.ts:40]: We should also check if `graphAdminRPCEndpoint ` and `ipfsEndpoint` are valid end points.

File: subgraph/src/DeployOriginSubgraph.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  erc20Gateway: string;
+  graphAdminRPCEndpoint: string;
+  ipfsEndpoint: string;
+} {
+  assert(process.argv.length === 6);
+
+  const originAnchor = process.argv[2];
+  const erc20Gateway = process.argv[3];
+  const graphAdminRPCEndpoint = process.argv[4];
+  const ipfsEndpoint = process.argv[5];
```
Comment (abhayks1) [inline subgraph/src/DeployOriginSubgraph.ts:41]: Same comment as above for validating graphAdminRPCEndpoint and ipfsEndpoint values

**Review by abhayks1 — CHANGES_REQUESTED** [review]
Good work 👍 
There are inline comments 📓

File: README.md (hunk 2 above)
Comment (0xsarvesh) [reply README.md:288]: The command existing in the parent node project. So readme exists at facilitator layer.

File: package.json (hunk 3 above)
Comment (0xsarvesh) [reply package.json]: Done

File: package.json (hunk 4 above)
Comment (0xsarvesh) [reply package.json]: Removed.

File: subgraph/src/DeployAuxiliarySubgraph.ts (hunk 5 above)
Comment (0xsarvesh) [reply subgraph/src/DeployAuxiliarySubgraph.ts]: Addd logger.

File: subgraph/src/DeployOriginSubgraph.ts (hunk 6 above)
Comment (0xsarvesh) [reply subgraph/src/DeployOriginSubgraph.ts]: Added logger.

File: subgraph/src/DeployAuxiliarySubgraph.ts (hunk 7 above)
Comment (0xsarvesh) [reply subgraph/src/DeployAuxiliarySubgraph.ts:40]: Added regex.

File: subgraph/src/DeployOriginSubgraph.ts (hunk 8 above)
Comment (0xsarvesh) [reply subgraph/src/DeployOriginSubgraph.ts:41]: Added regex.

**Review by abhayks1 — APPROVED** [review]

**Comment by abhayks1** [comment]
LGTM 👍

**Pull request merged**